google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/ErrorCountPerConnectionTest.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+
+import com.google.api.gax.core.FixedExecutorProvider;
+import com.google.bigtable.v2.*;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.FakeServiceBuilder;
+import com.google.cloud.bigtable.data.v2.models.*;
+import com.google.cloud.bigtable.stats.StatsRecorderWrapper;
+import io.grpc.*;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+@RunWith(JUnit4.class)
+public class ErrorCountPerConnectionTest {
+  private static final String SUCCESS_TABLE_NAME = "some-table";
+  private static final String ERROR_TABLE_NAME = "nonexistent-table";
+  private Server server;
+  private final FakeService fakeService = new FakeService();
+  private EnhancedBigtableStubSettings.Builder builder;
+  private ArgumentCaptor<Runnable> runnableCaptor;
+  private StatsRecorderWrapper statsRecorderWrapper;
+
+  @Before
+  public void setup() throws Exception {
+    server = FakeServiceBuilder.create(fakeService).start();
+
+    ScheduledExecutorService executors = Mockito.mock(ScheduledExecutorService.class);
+    builder =
+        BigtableDataSettings.newBuilderForEmulator(server.getPort())
+            .stubSettings()
+            .setBackgroundExecutorProvider(FixedExecutorProvider.create(executors))
+            .setProjectId("fake-project")
+            .setInstanceId("fake-instance");
+    runnableCaptor = ArgumentCaptor.forClass(Runnable.class);
+    System.out.println("rez1 " + executors + " 2 = " + builder);
```

Comment:
nit: clean up the debug log :P

Reply:
I had searched for all occurrences of "reza", but forgotten about "rez" 🤣